dali/python/nvidia/dali/data_node.py

```diff
@@ -67,6 +68,9 @@ def __str__(self):
 
     __repr__ = __str__
 
+    def __hash__(self) -> int:
+        return hash(str(self) + str(self.source))
```

Comment:
Should be, but thinking about it, maybe it would be better to use just the repr or string as the keys in the sets. I cannot use the DataNode directly, as it has the `__equal__` overloaded to return the arithmetic op.
I could use AutoGraph to replace the comparisons with something else, but I'm not convinced if it's not to many changes and differently working arithmetic ops depending on the mode.